yt_dlp/extractor/naver.py

```diff
@@ -247,3 +254,174 @@ def _extract_video_info(self, video_id, url):
+                note=f'Downloading JSON highlights for show {show_id} - page {page}')
+            highlight = [
+                v for v in highlights_videos.get('results', [])
+                if v.get('id', -1) == int(highlight_id)
+            ]
+            if highlight or highlights_videos.get('count', 0) <= self._PAGE_SIZE * (page + 1):
+                break
+            page += 1
+
+        if not highlight:
+            raise ExtractorError(f'Unable to find highlight {highlight_id} for show {show_id}')
+
+        highlight = highlight[0]
+        return merge_dicts(
+            {
+                'id': f'{show_id}-{highlight_id}',
+                'title': highlight.get('title'),
+                'timestamp': unified_timestamp(highlight.get('regdate')),
+                'thumbnail': highlight.get('thumbnail_url'),
+            },
+            self._extract_video_info(
+                highlight_id, highlight['video_id'], highlight['video_inkey']))
+
+    def _extract_show_replays(self, show_id):
+        page = 0
+        entries = []
+        while True:
+            show_vod_info = self._call_api(
+                f'/vod-shows/{show_id}', show_id,
+                query={'offset': page * self._PAGE_SIZE, 'limit': self._PAGE_SIZE},
+                note=f'Downloading JSON vod list for show {show_id} - page {page}'
+            ).get('response', {}).get('result', {})
+            for v in show_vod_info.get('vod_list', []):
+                entries.append(self._extract_replay(show_id, v['id']))
+
+            if show_vod_info.get('count', 0) <= self._PAGE_SIZE * (page + 1):
+                break
+            page += 1
+        return entries
+
+    def _extract_show_highlights(self, show_id):
+        page = 0
+        entries = []
+        while True:
+            highlights_videos = self._call_api(
+                f'/shows/{show_id}/highlights/videos/', show_id,
+                query={'offset': page * self._PAGE_SIZE, 'limit': self._PAGE_SIZE},
+                note=f'Downloading JSON highlights for show {show_id} - page {page}')
+
+            highlights = highlights_videos.get('results', [])
+            for v in highlights:
+                entries.append(self._extract_highlight(show_id, v['id'], highlights_videos))
+
+            if highlights_videos.get('count', 0) <= self._PAGE_SIZE * (page + 1):
+                break
+            page += 1
+        return entries
+
+    def _real_extract(self, url):
+        show_id = self._match_id(url)
+        qs = compat_parse_qs(compat_urllib_parse_urlparse(url).query)
+
+        if qs.get('shareHightlight') and not self._yes_playlist(show_id, qs["shareHightlight"][0]):
+            return self._extract_highlight(show_id, qs['shareHightlight'][0])
+
+        if qs.get('shareReplayId') and not self._yes_playlist(show_id, qs["shareReplayId"][0]):
+            return self._extract_replay(show_id, qs['shareReplayId'][0])
```

Comment:
it is used here separately

Reply:
ah, u r right. I misunderstood the code

Reply:
Even if pagedlist arent usefull here, it should atleast be a generator. Dont create+append lists

Reply:
yes, sorry